### ISSUE TITLE
Color contrast issues against red background

### ISSUE THREAD

**Issue by vperezcadavid99** [phetsims/resistance-in-a-wire#234]
**Test device**
Asus Zenbook

**Operating System**
Win 11

**Browser**
Firefox

**Problem description**
From https://github.com/phetsims/qa/issues/1399

When the background is completely red:

- The color contrast fails for the blue variables: resistivity, length, and area. Contrast ratio: 2.12:1
- The color contrast fails for the reset all button (even with the dark border). Contrast ratio: 1.51:1

**Visuals**

<img width="257" height="235" alt="Image" src="https://github.com/user-attachments/assets/740dfd56-f361-4df9-b693-3f9a083aba81" />

<img width="162" height="176" alt="Image" src="https://github.com/user-attachments/assets/63c3faa2-9432-4092-b003-79d9e079f0b9" />

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Resistance in a Wire‬
URL: https://phet-dev.colorado.edu/html/resistance-in-a-wire/1.7.0-dev.11/phet/resistance-in-a-wire_all_phet.html
Version: 1.7.0-dev.11 2026-05-15 21:05:48 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/148.0.0.0 Safari/537.36
Language: en-US
Window: 1600x812
Pixel Ratio: 1.7999999523162842/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}
<details/>

**Comment by terracoda** [comment]
Nice catch @vperezcadavid99. This sim predates more robust color contrast testing.
@arouinfar, are there any default practices like existing changes to the reset all button, or do we need something custom?

@vperezcadavid99 , is it just the tiny A that is not passing, or is it the color itself that is not passing?

**Comment by terracoda** [comment]
You listed the numbers :-) Thanks.

**Comment by terracoda** [comment]
@arouinfar, do you think a white stroke could work in these types of scenarios? Or a stroke of the BG color, so it is not perceptible when the letters are over the regular BG?

**Comment by arouinfar** [comment]
Good catch @vperezcadavid99!

> The color contrast fails for the blue variables: resistivity, length, and area. Contrast ratio: 2.12:1

I think the best way to address this would be to add a stroke around the variables that matches the background color of the sim,  as @terracoda suggested. @jessegreenberg can you please take care of that?

> The color contrast fails for the reset all button (even with the dark border). Contrast ratio: 1.51:1

The success criterion for buttons requires that the button's icon/text has sufficient contrast against the button background (which it does as of https://github.com/phetsims/scenery-phet/issues/938). The contrast ratio between the button border/background and the sim background is irrelevant, so no changes are needed.

**Comment by jessegreenberg** [comment]
The stroke around Text actually eats into the letter shape, so just adding a stroke makes letters look like this:

<img width="445" height="307" alt="Image" src="https://github.com/user-attachments/assets/27c4e839-14bb-4e7a-8b03-e4f722f6d554" />

Ill see if a more custom rendering solution might work (stacking Text to create this effect).

**Comment by arouinfar** [comment]
@jessegreenberg another alternative would be to add a rectangle behind each letter (or the entire right side of the equation) that matches the background color of the sim, but with an opacity of 50% or so.

**Comment by jessegreenberg** [comment]
If we use a "stacked" letter - it is better, but the created halo does not overlap perfectly, here is the result:

<img width="370" height="235" alt="Image" src="https://github.com/user-attachments/assets/799f4b53-0d7f-4c59-94c2-b1d17ae7a231" />

**Comment by jessegreenberg** [comment]
If the outline width is 0.2, it is not as noticeable:

<details>

<img width="450" height="340" alt="Image" src="https://github.com/user-attachments/assets/419325b2-0050-4a1a-bafd-e4184e718d5d" />

</details>

Here is what it looks like trying a background recgtangle. We can adjust the dilation, but it is more strange when right hand side letters overlap:

<details>

<img width="431" height="335" alt="Image" src="https://github.com/user-attachments/assets/3a8b5961-b52d-49d3-bb16-549f0ddba16a" />

</details>

Scenery supports a drop shadow, here is what that would look like this:

<details>

<img width="390" height="325" alt="Image" src="https://github.com/user-attachments/assets/2edc6e57-75df-4dd3-b823-df42bb270267" />

</details>

That was too subtle, so this is what stacking multiple shadows looks like. This is 8 shadows with offsets per character. However, performance is too bad even on my very fast machine:

<details>

<img width="450" height="325" alt="Image" src="https://github.com/user-attachments/assets/366b32d2-a9fb-4425-9391-54f80250083d" />

</details>


@arouinfar can you please take a look at these options? What would you like to pursue? Or do you have other designs you want to try?

**Comment by jessegreenberg** [comment]
Sorry, I overlooked your recommendation to add a rectangle to the right hand side, seeing what that looks like, one sec...

**Comment by jessegreenberg** [comment]
A layering rectangle like this:

<details>

<img width="400" height="300" alt="Image" src="https://github.com/user-attachments/assets/d7e9e407-ee10-4c72-86b0-b01ae28fbd9b" />

</details>

And:

<details>

<img width="400" height="250" alt="Image" src="https://github.com/user-attachments/assets/bee1a4f4-b085-47be-a6f7-fa221ad8383b" />

</details>

Or, we could try changing the colors entirely?

**Comment by jessegreenberg** [comment]
On more idea...layered background text with a very small offset:

<details>

<img width="433" height="323" alt="Image" src="https://github.com/user-attachments/assets/8f3d3bf6-f290-483f-81c8-d6b10e467bba" />

</details>

**Comment by arouinfar** [comment]
@jessegreenberg thanks for prototyping so many options! None of the layered rectangle options look good to me, so I think we should pursue one of the options that replicates outlining the text.

The stacked drop shadows look the best to me, but the performance problems make it a no-go.

> If the outline width is 0.2, it is not as noticeable:
> <img width="906" height="680" alt="Image" src="https://github.com/user-attachments/assets/3ccda911-a7b8-407a-8872-823499f8e021" />

This version of the stacked letters looks best, so let's go with that. The A is a bit hard to see, but I think that's mostly due to its size. Extreme sizes are important the learning goals of the simulation, anyway.

**Comment by jessegreenberg** [comment]
Thanks @arouinfar , can you please confirm that it is working well to solve this problem in the running sim?

**Comment by arouinfar** [comment]
Thanks @jessegreenberg, the changes look good on main, and I believe they adequately address the contrast issue. I think we're good to go here, closing.
